src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -52,4 +55,19 @@ public CinemaHall add(CinemaHall cinemaHall) {
             throw new DataProcessingException("Can't get all cinema hall", e);
         }
     }
+
+    @Override
+    public CinemaHall getByCinemaHallId(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<CinemaHall> criteriaQuery =
+                    criteriaBuilder.createQuery(CinemaHall.class);
+            Root<CinemaHall> root = criteriaQuery.from(CinemaHall.class);
+            Predicate predicateForId = criteriaBuilder.equal(root.get("id"), id);
+            criteriaQuery.where(predicateForId);
+            return session.createQuery(criteriaQuery).uniqueResult();
```

Comment:
```suggestion
          return session.get(CinemaHall.class, id);
```
try this one(in other daos too)